docs/wagtail-migrations.md

```diff
@@ -1,12 +1,97 @@
+## Table of contents
 
-To this end, there are some utility functions in cfgov-refresh that make this easier. Using these utilities, a Django data migration that modifies a StreamField would use the following format:
+1. [Reference material](#reference-material)
+1. [Schema migrations](#schema-migrations)
+1. [Data migrations](#data-migrations)
+   1. [Utility functions](#utility-functions)
+
+
+## Reference material
+
+The following links may be useful for setting context or diving deeper
+into the concepts presented throughout this page:
+
+- [Django migrations documentation](https://docs.djangoproject.com/en/1.11/topics/migrations/)
+- [Django data migrations documentation](https://docs.djangoproject.com/en/1.11/topics/migrations/#data-migrations)
+- [Wagtail Streamfield migrations documentation](https://docs.wagtail.io/en/v1.13.4/topics/streamfield.html#migrations)
+
+
+## Schema migrations
+
+Any time you are working with StreamField block or Wagtail page definitions,
+a Django schema migration will be required.
+
+To automatically generate a schema migration,
+run the following, editing it to give your migration a name
+that briefly describes the change(s) you're making:
+
+```bash
+./cfgov/manage.py makemigrations -n <description_of_changes>
+```
+
+It is a good idea to wait to do this until just before merging your new code
+into the `master` branch, to avoid potential conflicts with other developers'
+migrations that might make it into `master` before yours.
+Be aware, though, that our [back-end tests that run in Travis](../travis/)
+will fail if a required schema migration is missing.
+
+!!! note
+    Some changes will generate multiple migration files.
+    If you change a block that is used in pages defined in different sub-apps,
+    you will see a migration file for each of those sub-apps.
+
+
+## Data migrations
+
+If you…
+
+- … are renaming an existing field …
+- … are deleting a field (or converting it to a new type of field)
+  and don't want to lose any existing data stored in that field …
+- … are renaming a block within a page's StreamField definition …
+- … are deleting a block (or converting it to a new type of block)
+  and don't want to lose any existing data stored in that field …
+
+… then you will need a data migration!
+
+In other words, if an existing field is changing,
+any data stored in that field has to be migrated to the new field,
+unless you're OK with jettisoning it.
```

Comment:
The use of "jettisoning" implies that the data will be lost, which isn't necessarily true. It will still exist in the database (at least until the next time the page is saved).

Reply:
It depends on what the migrations are doing. If the "field" in question is a regular model field, then the schema migration is deleting that column from the database, so that data is lost.

It's only the quirk of StreamFields being stored as JSON that means *their* fields are the most special of snowflakes.

Reply:
Still in the DB or no, it's probably safest to encourage people to think of the data as lost, no?

Reply:
I agree 💯%. Data that only exists until the next time an object is saved is data that might as well be treated as lost.